traits/tests/test_callable.py

```diff
@@ -90,12 +90,47 @@ def test_callable_in_complex_trait(self):
                 a.callable_or_str = value
             self.assertEqual(a.callable_or_str, old_value)
 
+    def test_disallow_none(self):
+
+        class MyNewCallable(HasTraits):
+            value = Callable(default_value=pow, allow_none=False)
+
+        obj = MyNewCallable()
+
+        self.assertIsNotNone(obj.value)
+
+        with self.assertRaises(TraitError):
+            obj.value = None
+
```

Comment:
This may be overkill, but I think it's worth adding another check that `obj.value` is not `None` at this point. (It's conceivable that the implementing code could set `None` and _then_ raise the `TraitError`.)

Reply:
Whoops; sorry. Your next line does effectively check that `obj.value` is unchanged. Please ignore the above comment.